### ISSUE TITLE
Evaluation results

### ISSUE THREAD

**Issue by ed-fish** [rishikksh20/ViViT-pytorch#1]
Hi,

Thanks for your work making a Pytorch version of the paper - much appreciated!

How does this implementation compare to results in the original paper. Specifically on the Moments in Time dataset. 

Thanks,

Ed

**Comment by marco-hmc** [comment]
I am also interested at this topic. 
If there is anyone could provide me more information about the model parameters that might help me fix the problem, I would be thankful for that because using the default parameters is always overfitting.

Thanks.
Marco

**Comment by Linwei94** [comment]
I run the model on a very small dataset(51 classes with 20 video clips per class) and the result is very strange. Always output the same prediction. I wonder if it can be better if I load the pre-trained weights. I would appreciate anyone who can give me some tips. 

Thanks,
Dylan

**Comment by marco-hmc** [comment]
> I run the model on a very small dataset(51 classes with 20 video clips per class) and the result is very strange. Always output the same prediction. I wonder if it can be better if I load the pre-trained weights. I would appreciate anyone who can give me some tips.
> 
> Thanks,
> Dylan

I am facing with the same situation with you. I still don't have any ideat about that now. Waiting for the reply for authors.

**Comment by Linwei94** [comment]
> > I run the model on a very small dataset(51 classes with 20 video clips per class) and the result is very strange. Always output the same prediction. I wonder if it can be better if I load the pre-trained weights. I would appreciate anyone who can give me some tips.
> > Thanks,
> > Dylan
> 
> I am facing with the same situation with you. I still don't have any ideat about that now. Waiting for the reply for authors.

I wonder whether the problem results from the code or from my too-small dataset.

**Comment by vaibhavsah** [comment]
I tried it with a nearly 2000 Videos. Run different Epochs. But still the accuracy is not more than 21.09%. Strange thing is that it's same for most of the runs. No change in figures.

**Comment by Linwei94** [comment]
> I tried it with a nearly 2000 Videos. Run different Epochs. But still the accuracy is not more than 21.09%. Strange thing is that it's same for most of the runs. No change in figures.

Your dataset is too small. You can try run ViViT with ViT’s weight loaded for both temporal and spatial part.

**Comment by vaibhavsah** [comment]
@DylanTao94 Can you share how I can do that.

**Comment by Linwei94** [comment]
Sry mate, my code is not allowed to share. You can follow the steps in ViViT paper.

**Comment by seandatasci** [comment]
yes this model works fine i've tested it on a dataset of 50k videos

**Comment by vaibhavsah** [comment]
@seandatasci i think I might be doing something wrong with the code. Can you help me out here. My code is [here](https://colab.research.google.com/drive/1eLpKn35kX70lAryWWBjCLbzA4JDCzbrr?usp=sharing)

**Comment by Mark-Dou** [comment]
> @seandatasci i think I might be doing something wrong with the code. Can you help me out here. My code is [here](https://colab.research.google.com/drive/1eLpKn35kX70lAryWWBjCLbzA4JDCzbrr?usp=sharing)

i have the same problems with you, and i wonder you have resolved the problems whether or not, the acc or auc results is lower than 50%, the dataset size is also 2000, Thank u

**Comment by mx-mark** [comment]
Inspired from the implementation of the ViViT by the author, we have reimplement the TimeSformer and ViViT, and release the pretrain-model weights on Kinetics600 can be found [here](https://github.com/mx-mark/VideoTransformer-pytorch)

**Comment by mnauf** [comment]
The model isn't learning. Trained on 2 classes of UCF101 dataset. Adam optimizer, CrossEntropyLoss
